lib/puppet/util/posix.rb

```diff
@@ -186,7 +186,11 @@ def get_posix_value(location, id_field, field)
     end
 
     if check_value != field
-      check_value_id = get_posix_field(location, id_field, check_value)
+      begin
+        check_value_id = get_posix_field(location, id_field, check_value)
+      rescue Puppet::DevError
+        return id
```

Comment:
Silently catching exceptions should be avoided as may hide different issues and investigating them is hard